File: mobile/lib/src/models/room.dart
class Room {
  Room({
    required this.id,
    required this.dormId,
    required this.floorId,
    required this.roomNumber,
    required this.capacity,
    required this.status,
    required this.occupancy,
  });

  final int id;
  final int dormId;
  final int floorId;
  final String roomNumber;
  final int capacity;
  final String status;
  final int occupancy;

  factory Room.fromJson(Map<String, dynamic> json) {
    return Room(
      id: json['id'] as int,
      dormId: json['dorm_id'] as int,
      floorId: json['floor_id'] as int,
      roomNumber: json['room_number'] as String? ?? '',
      capacity: json['capacity'] as int? ?? 0,
      status: json['status'] as String? ?? 'AVAILABLE',
      occupancy: json['occupancy'] as int? ?? 0,
    );
  }
}

File: backend/app/Models/Room.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\SoftDeletes;

class Room extends Model
{
    use HasFactory;
    use SoftDeletes;

    protected $fillable = [
        'dorm_id',
        'floor_id',
        'room_number',
        'capacity',
        'status',
    ];

    public function dorm()
    {
        return $this->belongsTo(Dorm::class);
    }

    public function floor()
    {
        return $this->belongsTo(Floor::class);
    }

    public function assignments()
    {
        return $this->hasMany(RoomAssignment::class);
    }

    public function activeAssignments()
    {
        return $this->hasMany(RoomAssignment::class)->where('active', true);
    }
}
